lib/java_buildpack/framework/fusion_reactor.rb

```diff
@@ -0,0 +1,123 @@
+# Cloud Foundry Java Buildpack
+# Copyright 2017 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'java_buildpack/component/versioned_dependency_component'
+require 'java_buildpack/framework'
+
+module JavaBuildpack
+  module Framework
+    # Encapsulates the functionality for enabling FusionReactor support.
+    class FusionReactor < JavaBuildpack::Component::VersionedDependencyComponent
+
+      def initialize(context, &version_validator)
```

Comment:
This should look a bit more like the [New Relic integration](https://github.com/cloudfoundry/java-buildpack/blob/master/lib/java_buildpack/framework/new_relic_agent.rb#L24).  There's no need to do all of this assignment here; it can be done lazily once the decision to compile or release has been made.